### PR TITLE
added ts config

### PR DESCRIPTION
**Pull request by saiumesh535**
1. added ts config and written files in TS.

### CODE REVIEW

File: validation.js [hunk 1]
```diff
@@ -1,54 +1,56 @@
...
-const validator = {
-  min: (input, value) => input.length >= value,
-  max: (input, value) => input.length <= value,
-  type: (input, value) => {
-    if (!typedValidations[value]) {
-      return 1;
-    }
-    return typedValidations[value](input)
-  },
+var validator = {
```
Comment (lava-dukanam) [inline validation.js:41]: why was this changed from const to var?

File: validation.js [hunk 2]
```diff
@@ -1,54 +1,56 @@
...
 };
-
 /* typed validations */
-const typedValidations = {
-  email: (input) => /^([\w\!\#$\%\&\'\*\+\-\/\=\?\^\`{\|\}\~]+\.)*[\w\!\#$\%\&\'\*\+\-\/\=\?\^\`{\|\}\~]+@((((([a-z0-9]{1}[a-z0-9\-]{0,62}[a-z0-9]{1})|[a-z])\.)+[a-z]{2,6})|(\d{1,3}\.){3}\d{1,3}(\:\d{1,5})?)$/i.test(input), // example validation you can write your own regex
-  number: (input) => !isNaN(input),
-  boolean: (input) => input.toLowerCase() === ('true' || 'false') ? true : false,
-}
\ No newline at end of file
+var typedValidations = {
```
Comment (lava-dukanam) [inline validation.js:52]: why was this changed from const to var?

File: validation.js [hunk 3]
```diff
@@ -1,54 +1,56 @@
...
-    const isValid = validator[value] && validator[value](input[key], config[key][value]);
-    if (!isValid) {
-      const errorType = value === 'type' ? config[key][value] : value
-      return [...cumm, { for: key, errorType, value: config[key][value] }]
-    } else {
-      return isValid === 1 ? [...cumm, { for: key, message: `missing type config for ${key}` }] : cumm;
-    }
-  }, []);
-  return validation;
+    var validation = Object.keys(config[key]).reduce(function (cumm, value) {
```
Comment (lava-dukanam) [inline validation.js:29]: why was this changed from const to var?
can we switch to fat arrow syntax?

**Review by lava-dukanam — CHANGES_REQUESTED** [review]
added some minor comments

File: validation.js (hunk 1 above)
Comment (saiumesh535) [reply validation.js:41]: compiled file. we will use webpack to do this as discussed.

**Comment by saiumesh535** [comment]
closed, we will implement later.

**Pull request closed without merging**